src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ShardIndexingPressureMetricsCollector.java

```diff
@@ -0,0 +1,218 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.index.IndexingPressure;
+import org.jooq.tools.json.JSONObject;
+import org.jooq.tools.json.JSONParser;
+import org.jooq.tools.json.ParseException;
+
+import java.lang.reflect.Field;
+import java.util.Map;
+
+public class ShardIndexingPressureMetricsCollector extends PerformanceAnalyzerMetricsCollector implements MetricsProcessor {
+    public static final int SAMPLING_TIME_INTERVAL = MetricsConfiguration
+        .CONFIG_MAP.get(ShardIndexingPressureMetricsCollector.class).samplingInterval;
+    private static final int KEYS_PATH_LENGTH = 0;
+    private static final Logger LOG = LogManager.getLogger(ShardIndexingPressureMetricsCollector.class);
+    private static final ObjectMapper mapper = new ObjectMapper();
+    private static final JSONParser parser = new JSONParser();
+
+    //Class And Field Names
+    public static final String SHARD_INDEXING_PRESSURE_CLASS_NAME = "org.elasticsearch.index.ShardIndexingPressure";
+    public static final String CLUSTER_SERVICE_CLASS_NAME = "org.elasticsearch.cluster.service.ClusterService";
+    public static final String INDEXING_PRESSURE_CLASS_NAME = "org.elasticsearch.index.IndexingPressure";
+    public static final String SHARD_INDEXING_PRESSURE_STORE_CLASS_NAME = "org.elasticsearch.index.ShardIndexingPressureStore";
+    public static final String INDEXING_PRESSURE_FIELD_NAME = "indexingPressure";
+    public static final String SHARD_INDEXING_PRESSURE_FIELD_NAME = "shardIndexingPressure";
+    public static final String SHARD_INDEXING_PRESSURE_STORE_FIELD_NAME = "shardIndexingPressureStore";
+    public static final String SHARD_INDEXING_PRESSURE_HOT_STORE_FIELD_NAME = "shardIndexingPressureHotStore";
+
+    private final ConfigOverridesWrapper configOverridesWrapper;
+    private final PerformanceAnalyzerController controller;
+    private StringBuilder value;
+
+    public ShardIndexingPressureMetricsCollector(PerformanceAnalyzerController controller,
+                                                 ConfigOverridesWrapper configOverridesWrapper) {
+        super(SAMPLING_TIME_INTERVAL, "ShardIndexingPressureMetricsCollector");
+        value = new StringBuilder();
+        this.configOverridesWrapper = configOverridesWrapper;
+        this.controller = controller;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void collectMetrics(long startTime) {
+        if(!controller.isCollectorEnabled(configOverridesWrapper, getCollectorName())) {
+            return;
+        }
+
+        try {
+            Class.forName(SHARD_INDEXING_PRESSURE_CLASS_NAME);
+        } catch (ClassNotFoundException e) {
+            LOG.debug("Shard IndexingPressure not present in this ES version. Skipping ShardIndexingPressureMetricsCollector");
```

Comment:
I am worried that for cases, when the plugin is not installed, we might fill up the logs. We should just emit the log once at first run.